### ISSUE TITLE
Chart-Scrolling not working when xAxis-Range is equal to range of initial data

### ISSUE THREAD

**Issue by nickwinger** [highcharts/highcharts#14742]
#### Expected behaviour
When adding new data with a given range, the chart should scroll and only show the range

#### Actual behaviour
When you set the range exactly like the initial data the scrolling does not work

#### Live demo with steps to reproduce
if you look here, the scrolling works because the range is smaller than the initial data:
(initial data is one minute, range is only 40 seconds):
https://stackblitz.com/edit/ionic-v4-angular-tabs-kcdmqy?file=src/app/tab1/tab1.page.ts

however if i set the range to one minute (which is desired), the chart gets squeezed when updating and does not scroll anymore:
https://stackblitz.com/edit/ionic-v4-angular-tabs-phr1tk?file=src/app/tab1/tab1.page.ts

#### Product version

latest

#### Affected browser(s)
safari/chrome macos

**Comment by pawelfus** [comment]
Hi @nickwinger 

Thank you for reporting this issue. Let me explain what happens: 
First,`axis.range` sets the range, then auto-scrolling is handled by Scrollbar/Navigator. Extremes set by Navigator/Scrollbar have higher priority than `axis.range`.

*(for the explanation I will use Navigator, but the same happens with Scrollbar)*
If the right handle of the Navigator is at the max pixel position and the dataset is updated, then extremes are changed to reflect changes in the dataset: new points added? Increase `max`. Some points removed? Decrease `max`. No changes at the end of the dataset? Don't change `max`.

If the left handle of the Navigator is somewhere in the middle of the dataset, then the current range is kept when adding/removing points at the end of the dataset (your first demo shows that).

If the left handle of the Navigator is at the min pixel position, then logic is exactly the same as for the right handle. Each of the handles is calculated separately, then extremes are set.

Just in case you wonder why it works that way: Imagine you are the user. You use Navigator to show the full range (or click on the RangeSelector button, or use any other solution that developers prepared, to manage extremes on the chart). But, right after new points arrive, extremes are changed to the ones you have just changed from. All because developers decided to set `xAxis.range` (or `xAxis.min` etc).

Instead of this, we let developers handle extremes as they want using `xAxis.setExtremes()`. For example, your second case, updated: https://stackblitz.com/edit/ionic-v4-angular-tabs-bfitwj?file=src%2Fapp%2Fchart%2Fchart.component.ts - but you will meet the same issue: if a user manually changes the extremes, how to detect it? We can use the `afterSetExtremes` event perhaps...

Anyway, if you have some ideas on how to resolve this, let us know! We will discuss this issue at the next meeting.

PS: Setting `axis.range` does not increase `min` or `max`: https://jsfiddle.net/BlackLabel/jzmgc9v4/1/

**Comment by nickwinger** [comment]
Hi Pawel,

i'm not sure if i got you on this one, also there is again lots of logic going on (overuling range etc.)

Personally i'm not a fan of "magic" happening, so it would be good to just have somekind of "mode" property to be userfriendly or something.
Also i'm not sure if i understand what the "problem" to resolve is.

I understand that when the user drage the range to the min and max that it is meant to be "all", and not a specifig range.
in that case i would just set the range to undefined or null, meaning show me everything
i'm thinking about the rangeSelector, where you have the option of "all".

however if i set the range to a specific value, then that value should be taken, regardless the min and max.

also i don't understand the "auto-scrolling" logic you mentioned.
what extremes are set by scrollbar or navigator ?

i think the rangeSelector is a good place to start:
there you have the type: 'all'
i think here it is crystal clear that i want everything to be shown, even when removing or adding points.
however when i set a specific range, i want that to be constant.
so e.g. if the user drags the handles to the min and max, then the range gets switched to the "all" type and it stays there
until a specific range is set.

i don't understand why auto-scrolling takes extremes ?!?
auto-scrolling should take the range, then it is clear if i want "all" or a specific range.

**Comment by pawelfus** [comment]
Sorry for the confusion, let me rephrase it:
RangeSelector and Navigator are extensions that modify what happens on xAxis. When you add points we check the range that Navigator holds. If this is a full range, we keep it. No matter the `xAxis.range` option.

Anyway, a workaround to achieve what you described: 
Demo: https://jsfiddle.net/BlackLabel/3dcz7sL1/
```js
(function(H) {
  H.wrap(
    H.Navigator.prototype,
    'modifyBaseAxisExtremes',
    function(proceed) {
      const chart = this.chart,
        nav = chart.navigator;

      if (
        Highcharts.defined(this.options.range) &&
        !Highcharts.defined(chart.fixedRange)
      ) {
        nav.stickToMin = this.options.range > this.max - this.min;
      }

      return proceed.apply(this, Array.prototype.slice.call(arguments, 1));
    }
  )
})(Highcharts);
```

**Comment by nickwinger** [comment]
Hi Pawel,

thanks, the fix is working.
However also if i have rangeSelector and Navigator disabled, it does not scroll:
https://stackblitz.com/edit/ionic-v4-angular-tabs-phr1tk?file=src/app/tab1/tab1.page.ts

i think it would be good, if the navigator (and scrollbar ?) have the same option as the rangeSelector: "all".
So if a user manually drags to max and min limits, the mode goes to "all".
And if i programmatically set min max on the rangeSelector, Navigator or scrollbar the mode goes off "all".

**Comment by pawelfus** [comment]
> i think it would be good, if the navigator (and scrollbar ?) have the same option as the rangeSelector: "all".
> So if a user manually drags to max and min limits, the mode goes to "all".
> And if i programmatically set min max on the rangeSelector, Navigator or scrollbar the mode goes off "all".

I might be missing something, but this is how it works internally. And this is the reason why you reported this bug in the first place. Could you elaborate?

**Comment by nickwinger** [comment]
well, only if the user drags to the min max (meaning his intention is the "all" mode) then it should switch to all.
BUT, if i programmatically set the extremes to the min max (with a given concrete value in the range options) then it should NOT switch to the all mode, but take the extremes as a value (like in the range selector, "all" or a given "value")

**Comment by pawelfus** [comment]
Ok, thank you for the explanation

**Comment by pawelfus** [comment]
We will fix both, Navigator and Scrollbar, so the `range` option won't be ignored when there are fewer points :)

**Comment by nickwinger** [comment]
nice !
